packages/@lwc/engine-core/src/framework/mutation-tracker.ts

```diff
@@ -34,16 +35,16 @@ export function componentValueObserved(vm: VM, key: PropertyKey, target: any = {
     }
 
     // The portion of reactivity that's exposed to signals is to subscribe a callback to re-render the VM (templates).
-    // We check check the following to ensure re-render is subscribed at the correct time.
+    // We check the following to ensure re-render is subscribed at the correct time.
     //  1. The template is currently being rendered (there is a template reactive observer)
     //  2. There was a call to a getter to access the signal (happens during vnode generation)
     if (
         lwcRuntimeFlags.ENABLE_EXPERIMENTAL_SIGNALS &&
         isObject(target) &&
         !isNull(target) &&
-        'value' in target &&
-        'subscribe' in target &&
-        isFunction(target.subscribe) &&
+        safeHasProp(target, 'value') &&
+        safeHasProp(target, 'subscribe') &&
+        isFunction((target as any).subscribe) &&
```

Comment:
```suggestion
        isFunction(target.subscribe) &&
```
Not necessary with better types for `safeHasProp`